.github/workflows/operator-integration-test.yml

```diff
@@ -0,0 +1,90 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+name: operator Integration Test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [ opened, reopened ]
+    branches:
+      - main
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  AmazonCloudWatchAgentOperatorIntegrationTest:
+    name: AmazonCloudWatchAgentOperatorIntegrationTest
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Start minikube
+        uses: medyagh/setup-minikube@master
+
+      - name: Deploy cert-manager to minikube
+        run:
+          kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.0/cert-manager.yaml
+
+      - name: Verify minikube and cert-manager
+        run: |
+          sleep 10
+          kubectl get pods -A
+
+      - name: Build image
+        run: |
+          eval $(minikube docker-env)
+          make container
+          docker images
+
+      - name: Deploy operator to minikube
+        run:
+          make deploy
+
+      - name: Test case for AmagonCloudWatchAgent pod creation
+        run: |
+          helm template --namespace amazon-cloudwatch -s templates/cloudwatch-agent-daemonset.yaml ./helm | kubectl apply --namespace amazon-cloudwatch -f -
+          sleep 60
+          kubectl describe pods -n amazon-cloudwatch
+          pod_name="$(kubectl get pods -n amazon-cloudwatch -l app.kubernetes.io/component=amazon-cloudwatch-agent,app.kubernetes.io/instance=amazon-cloudwatch.cloudwatch-agent -o=jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')"
+          if [ -z "$pod_name" ]; then
+            echo "Pod $pod_name is not running. Exiting with ERROR."
+            echo "Pod is not created test case: FAIL"
+          else
+            echo "Pod $pod_name is running."
+            echo "Pod is created test case: PASS"
+          fi
+
+      - name: Test default instrumentation
+        run: |
+          cd integration-tests/operator
```

Comment:
nit: I would call the directory `manifests` instead of `operator`. And call the file `sample-deployment.yaml` instead of `test-deployment.yaml` so its extensible for future tests.

Reply:
done!!